activerecord/lib/active_record/connection_handling.rb

```diff
@@ -45,7 +45,7 @@ module ConnectionHandling
     # The exceptions AdapterNotSpecified, AdapterNotFound and +ArgumentError+
     # may be returned on an error.
     def establish_connection(spec = nil)
-      raise RuntimeError, "Anonymous class is not allowed." unless name
```

Comment:
Happy to open up another PR with that -- wanted to try and keep this PR centered on a few fixes to Arthur's PR. :-)

I'll open up another PR and ping you on it for review.